### PR TITLE
feat: add frozen and locked

### PR DESCRIPTION
**Pull request by jayy-lmao**
*Issue #, if available:*

*Description of changes:*
This introduces the frozen and locked behaviors to cargo chef. 
See: https://doc.rust-lang.org/cargo/commands/cargo-build.html#manifest-options

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -123,6 +123,12 @@ pub struct Cook {
     /// Build offline.
     #[clap(long)]
     offline: bool,
+    /// Build locked. Throw if Cargo.lock is not up to date
+    #[clap(long)]
+    locked: bool,
+    /// Build frozen. Locked, but also don't access network
+    #[clap(long)]
+    frozen: bool,
```
Comment (LukeMathWalker) [inline src/main.rs:131]: Let's match the documentation strings from `cargo build` here.

**Comment by atwam** [comment]
Hi, just saw this PR as I was about to work on something similar, this time to pass on `--verbose` to the underlying cargo calls.
@jayy-lmao, if you mind including `verbose` the same way you have `frozen` and `locked` when you fix the comments as per @LukeMathWalker request, that would be great.
If not, no big deal, I'll take inspiration in your PR to submit a PR of my own.

**Comment by jayy-lmao** [comment]
> Hi, just saw this PR as I was about to work on something similar, this time to pass on `--verbose` to the underlying cargo calls. @jayy-lmao, if you mind including `verbose` the same way you have `frozen` and `locked` when you fix the comments as per @LukeMathWalker request, that would be great. If not, no big deal, I'll take inspiration in your PR to submit a PR of my own.

Just did this now, ty

File: src/main.rs [hunk 2]
```diff
@@ -123,6 +123,15 @@ pub struct Cook {
     /// Build offline.
     #[clap(long)]
     offline: bool,
+    /// Require Cargo.lock is up to date
+    #[clap(long)]
+    locked: bool,
+    /// Use verbose output (-vv very verbose/build.rs output)
```
[the remark this answers is not in the document]
Comment (atwam) [reply src/main.rs:129]: Probably yes, to keep things simple with a single `--verbose`. Not sure it's worth having the full compatibility with several levels of verbosity.

[the remark this answers is not in the document]
Comment (LukeMathWalker) [reply src/main.rs:129]: We can add support for it, but we should amend the docs here if we don't.

**Comment by vidhanio** [comment]
with the addition of these flags, would `--frozen` become the "default" way to use `cargo-chef` now? i don't see why someone wouldn't want their ci builds to be frozen and only read from `Cargo.lock`.

**Review by LukeMathWalker — APPROVED** [review]

**Comment by LukeMathWalker** [comment]
It depends—generally speaking, it would probably be alright to make `--frozen` the default, but it's likely to break some builds and I don't think the benefits outweight the risks here.

**Comment by vidhanio** [comment]
@LukeMathWalker i meant default more as in "recommended" configuration, like adding it to the README.

**Comment by LukeMathWalker** [comment]
I don't think it adds enough value and it would probably lead to people using it without understanding what it actually does (because they are copying from the example), which would in turn lead to open issues here.

**Pull request merged**